MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Entity.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\SearchCriteria\CollectionProcessor\FilterProcessor;
+
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor\CustomFilterInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\Framework\DB\Select;
+
+/**
+ * Custom filter to filter collection by entity type
+ */
+class Entity implements CustomFilterInterface
+{
+    private const TABLE_ALIAS = 'main_table';
+    private const TABLE_MEDIA_CONTENT_ASSET = 'media_content_asset';
```

Comment:
Since we are depending on the content table here, shouldn't we add MediaContent implementation as dependency on MediaGalleryUi?

Reply:
Static test says no (and it would probably fail if we add it)